pkg/JuliaInterface/src/convert.c

```diff
@@ -27,22 +28,27 @@ jl_value_t * julia_gap(Obj obj)
     if (obj == False) {
         return jl_false;
     }
+    BEGIN_GAP_SYNC();
     if (TNUM_OBJ(obj) >= FIRST_EXTERNAL_TNUM &&
         CALL_1ARGS(JULIAINTERFACE_IsJuliaWrapper, obj) == True) {
         obj = CALL_1ARGS(JULIAINTERFACE_JuliaPointer, obj);
         if (IS_JULIA_OBJ(obj)) {
+            END_GAP_SYNC();
             return GET_JULIA_OBJ(obj);
         }
         // to handle immediate integers, booleans, FFEs, IS_JULIA_FUNC, we
         // use recursion - but only for internal types; this prevents an
         // infinite recursion, e.g. if an object's JuliaPointer points to the
         // object itself
-        if (TNUM_OBJ(obj) < FIRST_EXTERNAL_TNUM)
+        if (TNUM_OBJ(obj) < FIRST_EXTERNAL_TNUM) {
+            END_GAP_SYNC();
```

Comment:
I really think I'd prefer if we use `BEGIN_GAP_SYNC` / `END_GAP_SYNC` symmetrically.

Alternatively, we could convert all this code to C++, and use RAII to take care of beginning and ending the GAP "sync". That would then also avoid bugs where somebody accidentally returns w/o calling END_GAP_SYNC.

Reply:
In the meantime, in this particular function, we could easily restructure the code to have a single `return` at the end, though, no?

Reply:
Yes, I've done that. I've also considered whether we should just use `GAP_SYNC` as the sole primitive, similar to the `GAP_THROW` rework in GAP proper, if we're going to always match them. But (1) that may produce scoping issues (see below for more) and (2) we may sometimes have to split the lock/unlock parts across functions (undesirable from a code cleanliness perspective, but it may become necessary).